### PR TITLE
Add StoreMy REST API

### PR DESCRIPTION
**Pull request by okucheruk**
This PR adds REST API for StoreMy cloud file storage service.

**Implemented APIs:**
- User registration, authentication and account operations.
- FileBrowser file operations.

### CODE REVIEW

**Comment by okucheruk** [comment]
@dkalinin, PTAL.

File: app/lib/storemy/api/user/AuthenticationService.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  /**
+   * Authenticates StoreMy user.
+   *
+   * @param {Object} userAuthenticationDetails A StoreMy user details object.
+   * @return {Promise<Object>} The response object.
+   */
+  signIn(userAuthenticationDetails) {
+    return this._httpRequestWorker
+    // .post('https://doe.cloud/storemy/api/v1/user/authentication', userAuthenticationDetails)
+        .get('https://doe.cloud/try.json');
```
Comment (dkalinin) [inline app/lib/storemy/api/user/AuthenticationService.js]: Please add logging or requests with the real URL.

Comment (okucheruk) [reply app/lib/storemy/api/user/AuthenticationService.js]: Logging added at HttpRequest level. 
Mocks removed.

**Comment by okucheruk** [comment]
@dkalinin, fixed, PTAL.

**Comment by okucheruk** [comment]
@dkalinin, fixed, PTAL.

**Pull request merged**